app/api/events.py

```diff
@@ -391,13 +394,29 @@ def before_get_object(self, view_kwargs):
                 else:
                     view_kwargs['id'] = None
 
+    def before_update_object(self, event, data, view_kwargs):
+        if data.get('original_image_url') and data['original_image_url'] != event.original_image_url:
```

Comment:
Also, please download the original_image_url and resave it and update the new location/link